controller/src/main/java/org/jetbrains/MavenController.java

```diff
@@ -1,59 +1,102 @@
 package org.jetbrains;
 
+
+import ch.epfl.scala.bsp4j.BuildClient;
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier;
+import ch.epfl.scala.bsp4j.CompileParams;
+import ch.epfl.scala.bsp4j.StatusCode;
+import ch.epfl.scala.bsp4j.TaskId;
+import ch.epfl.scala.bsp4j.TaskProgressParams;
 import lombok.extern.log4j.Log4j2;
 import org.apache.maven.shared.invoker.DefaultInvocationRequest;
 import org.apache.maven.shared.invoker.DefaultInvoker;
 import org.apache.maven.shared.invoker.InvocationRequest;
 import org.apache.maven.shared.invoker.InvocationResult;
 import org.apache.maven.shared.invoker.Invoker;
 import org.jetbrains.maven.server.EventPacket;
-
 import java.io.File;
 import java.io.IOException;
+import java.net.ServerSocket;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.Properties;
 
 @Log4j2
 public class MavenController {
-    private final String projectDirectory;
     Invoker invoker;
 
-    public MavenController(String projectDirectory) {
+    BuildClient client;
```

Comment:
Can this be `private` and `final`? In general it's good to keep them `private` and `final` to avoid accidental changes.